lib/batch_loader/executor_proxy.rb

```diff
@@ -6,10 +6,10 @@ class BatchLoader
   class ExecutorProxy
     attr_reader :default_value, :block, :global_executor
 
-    def initialize(default_value, &block)
+    def initialize(default_value, key, &block)
       @default_value = default_value
       @block = block
-      @block_hash_key = block.source_location
+      @block_hash_key = key || block.source_location
```

Comment:
Maybe I'm overthinking this, but I can see a case for wanting the key to be `key.to_s + block.source_location` (allows the key to be unique on a per loader basis, rather than codebase wide) and I cannot see a case for _not_ wanting it to be `key.to_s + block.source_location`

Reply:
Yeah, I think that's a great idea! 

Even though with metaprogramming (`eval`) `key` should be still unique through the whole codebase, in rest of the cases it is very useful and may help avoid potential collisions.